ca/ca.go

```diff
@@ -304,22 +303,15 @@ var ocspStatusToCode = map[string]int{
 // certificate profile. The certificate profile is identified by a hash to ensure an exact match even if
 // the configuration for a specific profile _name_ changes.
 //
+// Returns precertificate DER.
+//
 // [issuance cycle]: https://github.com/letsencrypt/boulder/blob/main/docs/ISSUANCE-CYCLE.md
-func (ca *certificateAuthorityImpl) issuePrecertificate(ctx context.Context, issueReq *capb.IssueCertificateRequest) (*capb.IssuePrecertificateResponse, error) {
+func (ca *certificateAuthorityImpl) issuePrecertificate(ctx context.Context, certProfile *certProfileWithID, issueReq *capb.IssueCertificateRequest) ([]byte, error) {
 	// issueReq.orderID may be zero, for ACMEv1 requests.
```

Comment:
Do we want to retain ACMEv1 support? If so, maybe we should copy this comment to the top of `issueCertificateForPrecertificate`. If not, let's start checking `orderID` or maybe add a ` TODO`.

Reply:
Yep, I'm gonna make this be required-nonzero as a separate PR. It breaks a surprising number of unittests, which weren't passing an orderID.